### DISCUSSION TITLE
[bug]: "failed to connect to all addresses - Connection refused;  grpc_status:14

### DISCUSSION THREAD

**Discussion by Haso04** [lightningnetwork/lnd#8648] [Troubleshooting]
### Background

Running Django in windows, simple web app with this GRPC code connecting to my Raspiblitz LND to create a stub and generate invoices and payments for users.

Been running for a long while fine, noting i had the earlier version of the code which referred to the "rpc_pb2" files and not the new "lightning_pb2". However I have now generated these ones.

I noticed the problem occurring from late March, early April 24.

### Your environment

* version of `lnd`
LND 0.15.4-beta 

* which operating system (`uname -a` on *Nix)
Raspiblitz

* version of `btcd`, `bitcoind`, or other backend
 bitcoin v23.0.0

* any other relevant environment details

### Steps to reproduce

Run simple python script from windows/django:

```
import grpc
import os
import lightning_pb2 as ln
import lightning_pb2_grpc as lnrpc
import codecs

os.environ["GRPC_SSL_CIPHER_SUITES"] = 'HIGH+ECDSA' 

with open(os.path.expanduser(<path to macaroon>), 'rb') as f:
    macaroon_bytes = f.read()
    macaroon = codecs.encode(macaroon_bytes, 'hex')

#embeds encrytpted cert and creds (combined_cert) into stubbed channel, rather than having to pass macaroon each time
def metadata_callback(context, callback):
    # for more info see grpc docs
    callback([('macaroon', macaroon)], None)

cert = open(os.path.expanduser(<path to tls.cer>), 'rb').read()
cert_creds = grpc.ssl_channel_credentials(cert)
channel = grpc.secure_channel('localhost:10009', combined_creds)
stub = lnrpc.LightningStub(channel)
lndrequest = ln.Invoice(value=100) 
lndresponse = stub.AddInvoice(lndrequest)
print("Invoice: ", lndresponse.payment_request)
```

### Expected behaviour

generate 100 sat invoice and print to screen

**Note1:** when i run this same code on the actual Raspi and it works fine.
**Note2:** I have Fully Noded installed on my phone (assuming that's grpc?) and that connects to my node (from internal LAN and externally too) and I can generate invoices fine - so there shouldn't be any issues with firewalls/routers/port forwarding etc right?

### Actual behaviour

Error: failed to connect to all addresses - Connection refused;  grpc_status:14

**Comment by guggero** [comment]
`localhost:10009` means: "The computer that I am running this script on", which is obviously wrong if you're running this on another machine. You need to find out what the IP address of the RaspiBlitz is and then use that instead of `localhost`.

**Reply by Haso04** [reply]
Thanks for the sanity check. I was initially troubleshooting my upgrade to the new rpc files and in the process have indeed brought in errors with connectivity references. 

I think I was under the false impression if I was port forwarding on my router to my Raspi LAN IP - and my Raspi was listening on 0.0.0.0:10009 then localhost in the script would be fine. Lesson learnt 😅

Thanks again 🙌
